src/main/java/org/breedinginsight/brapi/v2/BrAPIGermplasmController.java

```diff
@@ -105,19 +113,88 @@ public HttpResponse<Object> searchGermplasm(
+
+        Optional<Program> program = programService.getById(programId);
+        if(program.isEmpty()) {
+            log.warn("Program id: " + programId + " not found");
+            return HttpResponse.notFound();
+        }
+
+        ApiResponse<Pair<Optional<BrAPIGermplasmListResponse>, Optional<BrAPIAcceptedSearchResponse>>> brapiGermplasm;
+        brapiGermplasm = brAPIEndpointProvider
+                .get(programDAO.getCoreClient(program.get().getId()), GermplasmApi.class)
+                .searchGermplasmSearchResultsDbIdGet(searchResultId.toString(), queryParams.getPage(), queryParams.getPageSize());
+
+        return getObjectHttpResponse(brapiGermplasm, program.get().getKey());
+    }
+
+    private HttpResponse<Object> getObjectHttpResponse(ApiResponse<Pair<Optional<BrAPIGermplasmListResponse>, Optional<BrAPIAcceptedSearchResponse>>> brapiGermplasm,
+                                                       String programKey) throws ApiException {
         if (brapiGermplasm.getBody().getLeft().isPresent()) {
-            return HttpResponse.ok(brapiGermplasm.getBody().getLeft().get());
+            // convert dbIds to DeltaBreed UUID
+            BrAPIGermplasmListResponse response = brapiGermplasm.getBody().getLeft().get();
+            List<BrAPIGermplasm> germplasm = response.getResult().getData();
+            germplasm.forEach(g -> setDbIdsAndStripProgramKeys(g, programKey));
+            return HttpResponse.ok(response);
         } else if (brapiGermplasm.getBody().getRight().isPresent()) {
             return HttpResponse.ok(brapiGermplasm.getBody().getRight().get());
         } else {
             throw new ApiException("Expected search response");
         }
+    }
+
+    /**
+     * Keep dbIds in DeltaBreed UUID context and strip program keys from synonyms and pedigree string for Field Book display
+     * @param germplasm
+     * @param programKey
+     */
+    private void setDbIdsAndStripProgramKeys(BrAPIGermplasm germplasm, String programKey) {
+        String extRef = Utilities.getExternalReference(germplasm.getExternalReferences(), "breedinginsight.org")
+                .orElseThrow(() -> new IllegalStateException("No BI external reference found"))
+                        .getReferenceId();
+
+        germplasm.germplasmDbId(extRef);
+
+        // Remove program keys from synonyms
+        if (germplasm.getSynonyms() != null && !germplasm.getSynonyms().isEmpty()) {
+            for (BrAPIGermplasmSynonyms synonym: germplasm.getSynonyms()) {
+                String newSynonym = Utilities.removeProgramKey(synonym.getSynonym(), programKey, germplasm.getAccessionNumber());
+                synonym.setSynonym(newSynonym);
+            }
+        }
 
+        // Remove program keys from pedigree string
+        String strippedPedigree = Utilities.removeProgramKeyAndUnknownAdditionalData(germplasm.getPedigree(), programKey);
+        germplasm.setPedigree(strippedPedigree);
     }
```

Comment:
```suggestion
     private void batchProcessGermplasm(List<BrAPIGermplasm> germplasmList, String programKey) throws IllegalStateException {
        // Prepare a regex pattern for program key removal
        Pattern programKeyPattern = Pattern.compile(String.format("\\s*\\[%s-.*?\\]\\s*", programKey));

        germplasmList.parallelStream().forEach(germplasm -> {
            // Set dbId
            germplasm.germplasmDbId(Utilities.getExternalReference(germplasm.getExternalReferences(), "breedinginsight.org")
                    .orElseThrow(() -> new IllegalStateException("No BI external reference found"))
                    .getReferenceId());

            // Process synonyms
            if (germplasm.getSynonyms() != null) {
                germplasm.getSynonyms().forEach(synonym -> {
                    synonym.setSynonym(Utilities.removeProgramKey(synonym.getSynonym(), programKey, germplasm.getAccessionNumber()));
                });
            }

            // Process pedigree
            if (germplasm.getPedigree() != null) {
                germplasm.setPedigree(programKeyPattern.matcher(germplasm.getPedigree()).replaceAll(""));
            }
        });
    }
```

Reply:
Were you advocating not changing `removeProgramKeyAndUnknownAdditionalData` in this card? I could see that so as to not take a chance breaking other stuff but if so should probably make that clear in the comments and maybe still put the regex string in a common area.

Reply:
Yep. The code suggestion above is just adding the batch method and using it to replace the for loop. I was not thinking of deleting `removeProgramKeyAndUnknownAdditionalData` for the reason you mentioned.

Reply:
pushed changes